elementary/utils/dicts.py

```diff
@@ -0,0 +1,21 @@
+from typing import Dict, List
+
+
+# Flatten a nested dict by a given key, for example:
+# nested_dict = {"top1": "one", "nested": {"top1": "One", "top2": "Two"}}, flatten_by_key = "nested" -> {"top1": "One", "top2": "Two"}
+def flatten_dict_by_key(nested_dict: Dict, flatten_by_key: str) -> Dict:
+    flatten_dict = {**nested_dict, **nested_dict.get(flatten_by_key, {})}
```

Comment:
True, but pop on nested_dict which is a variable of the method will pop out the key from the origin dict as well (outside of the method) which can cause some big voodoo 😱
(this is actually breaking the test, because we run the same dict with different scenarios and this is editing the dict outside of the method calls)

We can do:
```python
def flatten_dict_by_key(nested_dict: Dict, flatten_by_key: str) -> Dict:
    nested_dict_copy = {**nested_dict}
    inner_dict = nested_dict_copy.pop(flatten_by_key, {})
    return {**nested_dict_copy, **inner_dict}
```
Which IMO is quite the same with a different order 🙂 

I can change to that if you prefer it though